ucp/_libs/utils.pyx

```diff
@@ -49,38 +45,40 @@ def get_buffer_nbytes(buffer, check_min_size, cuda_support):
     if `check_min_size` is greater than the size of the buffer
     """
 
-    iface = None
-    if hasattr(buffer, "__cuda_array_interface__"):
-        iface = buffer.__cuda_array_interface__
-        if not cuda_support:
-            msg = "UCX is not configured with CUDA support, please add " \
-                  "`cuda_copy` and/or `cuda_ipc` to the UCX_TLS environment" \
-                  "variable and that the ucx-proc=*=gpu package is " \
-                  "installed. See " \
-                  "https://ucx-py.readthedocs.io/en/latest/install.html for " \
-                  "more information."
-            raise ValueError(msg)
-    elif hasattr(buffer, "__array_interface__"):
-        iface = buffer.__array_interface__
+    iface = getattr(buffer, "__array_interface__", None)
+    if iface is None:
+        iface = getattr(buffer, "__cuda_array_interface__", None)
```

Comment:
Similarly can we swap these?

Reply:
We need to check whether we have a CUDA object and whether those are allowed in this case below.

Reply:
But what happens if an array includes both `__array_interface__` and `__cuda_array_interface__`? It seems that the latter would never be used, though I don't know if there are any arrays that implement both.

Reply:
What would happen if an array implements both? It seems that then we would never try `__cuda_array_interface__`, though I don't know if this is valid and if some array type implements both.

Reply:
Sorry, I thought I thought I had lost my first comment, ignore the second which is basically the same.

Reply:
Do you know of a case where this would happen? TBH I don’t know of any libraries outside of NumPy that use `__array_interface__`. Most libraries rely on the Python Buffer Protocol (typically used via `memoryview`s).

Reply:
As NumPy also supports the Python Buffer Protocol, it might be worth dropping the `__array_interface__` handling here in favor of just routing that through a `memoryview` also.

Reply:
If the check for `__array_interface__` is exclusive to find whether it's a NumPy array, shouldn't we make it more clear by just checking `isinstance(buffer, np.ndarray)`? The `memoryview` idea makes sense as well, but it remains the question as to what order we should use to do the checking, `__cuda_array_interface__` or `memoryview` first.

Reply:
If we do `memoryview`, we just drop the `__array_interface__` check. `memoryview` is already checked after `__cuda_array_interface__`.

Reply:
Went ahead and switched NumPy to `memoryview` case in recent commits. So marking this resolved. Though if there are additional thoughts please feel free to start a new thread.